src/wallet/rpc/wallet.cpp

```diff
@@ -823,6 +823,102 @@ static RPCHelpMan migratewallet()
     };
 }
 
+static RPCHelpMan createwalletdescriptor()
+{
+    return RPCHelpMan{"createwalletdescriptor",
+        "Creates the wallet's descriptor for the given address type. "
+        "Only works on wallets that contain automatically generated descriptors. "
+        "The address type must be one that the wallet does not already have a descriptor for."
+        + HELP_REQUIRING_PASSPHRASE,
+        {
+            {"address_type", RPCArg::Type::STR, RPCArg::Optional::NO, "The address type the descriptor will produce. Options are \"legacy\", \"p2sh-segwit\", \"bech32\", and \"bech32m\"."},
+            {"internal", RPCArg::Type::BOOL, RPCArg::DefaultHint{"Both external and internal will be generated unless this parameter is specified"}, "Whether to only make one descriptor that is internal (if parameter is true) or external (if parameter is false)"},
```

Comment:
d13ebb5a43733a61245cdf396304c3ff67b64900: slight preference to drop this. It's confusing and I personally don't see any use for it. For advanced use case it seems better to me if users to manually construct the descriptor (with help from `getxpub` #22341). But maybe I missed something.

Reply:
I am ambivalent but I think it is a useful distinction as it can be used to rotate only specific descriptors rather than complete sets.